sui/gateway.js

```diff
@@ -150,11 +151,47 @@ async function callContract(keypair, client, config, chain, args, options) {
 }
 
 async function approveMessages(keypair, client, config, chain, args, options) {
-    if (!chain.contracts.axelar_gateway) {
+    if (!chain.contracts.AxelarGateway) {
         throw new Error('Axelar Gateway package not found.');
     }
 
-    const contractConfig = chain.contracts.axelar_gateway;
+    const contractConfig = chain.contracts.AxelarGateway;
+    const packageId = contractConfig.address;
+    const [multisigSessionId] = args;
+    const { payload, status } = await getMultisigProof(config, chain.axelarId, multisigSessionId);
+
+    if (!payload.messages) {
+        throw new Error('No messages to approve');
+    }
+
+    if (!status.completed) {
+        throw new Error('Multisig session not completed');
+    }
+
+    const executeData = executeDataStruct.parse(new Uint8Array(arrayify('0x' + status.completed.execute_data)));
+
+    const tx = new Transaction();
+
+    tx.moveCall({
+        target: `${packageId}::gateway::approve_messages`,
+        arguments: [
+            tx.object(contractConfig.objects.Gateway),
+            tx.pure(bcs.vector(bcs.u8()).serialize(new Uint8Array(executeData.payload)).toBytes()),
+            tx.pure(bcs.vector(bcs.u8()).serialize(new Uint8Array(executeData.proof)).toBytes()),
+        ],
+    });
+
+    await broadcast(client, keypair, tx);
+
+    printInfo('Approved messages');
```

Comment:
could you add the digest to the second argument? 

Here's a short snippet to get the digest.
```ts
const receipt = await broadcast(...);

printInfo('Approved messages', receipt.digest);
```


it's helpful for inspecting the submitted transaction in more details. Maybe apply this to other functions as well.